lib/ansible/modules/monitoring/zabbix/zabbix_user_group.py

```diff
@@ -0,0 +1,370 @@
+    login_password: password
+    state: present
+    name: Test group
+    rights:
+        - host_group: Admin VMs
+          permission: RW
+    users:
+        - user1
+        - user2
+'''
+
+RETURN = '''
+---
+template_json:
+  description: The JSON dump of the group
+  returned: when state is dump
+  type: str
+  sample: {
+    "group_json": {
+        "debug_mode": "0",
+        "gui_access": "0",
+        "name": "Test group",
+        "rights": [
+            {
+                "id": "20",
+                "permission": "3"
+            },
+        ],
+        "users": [
+            {
+                "userid": "6"
+            },
+            {
+                "userid": "12"
+            }
+        ],
+        "users_status": "0",
+        "usrgrpid": "19"
+    }
+  }
+'''
+
+import traceback
+
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+
+try:
+    from zabbix_api import ZabbixAPI, ZabbixAPIException
+    HAS_ZABBIX_API = True
+except ImportError:
+    ZBX_IMP_ERR = traceback.format_exc()
+    HAS_ZABBIX_API = False
+
+
+class UserGroup(object):
+    def __init__(self, module, zbx):
+        self._module = module
+        self._zapi = zbx
+
+    def get_group_id(self, group_name):
+        group_id = None
+        groups = self._zapi.usergroup.get({'filter': {'name': group_name}})
+        if len(groups) == 1:
+            group_id = groups[0]['usrgrpid']
+        return group_id
+
+    def delete_group(self, group_id):
+        if self._module.check_mode:
+            self._module.exit_json(changed=True)
+        self._zapi.usergroup.delete([group_id])
+
+    def dump_group(self, group_id):
+        groups = self._zapi.usergroup.get({'output': 'extend', 'usrgrpids': group_id, 'selectRights': 1, 'selectUsers': 1})
+        return groups[0]
+
+    def generate_group_config(self, name, debug_mode, gui_access, users_status, rights, users):
+        if debug_mode:
+            debug_mode = 1
+        else:
+            debug_mode = 0
+
+        gui_acesses = {'default': '0', 'internal': '1', 'ldap': '2', 'disabled': '3'}
+        gui_access = gui_acesses[gui_access]
+
+        if users_status == 'enabled':
+            users_status = 0
+        else:
+            users_status = 1
+
+        new_rights = []
+        for right in rights:
+            groups = self._zapi.hostgroup.get({'filter': {'name': right['host_group']}})
+            if not groups:
+                self._module.fail_json(msg="Target host group %s not found" % right['host_group'])
+            group_id = groups[0]['groupid']
+            permission = "0"
+            if right['permission'] == 'RO':
+                permission = "2"
+            elif right['permission'] == 'RW':
+                permission = "3"
+
+            new_rights.append({'id': group_id, 'permission': permission})
+
+        userids = []
+        for user in users:
+            user_array = self._zapi.user.get({'filter': {'name': user}})
```

Comment:
Using `name` as an identifier for users can lead to a lots of confusion. Imagine that I have three users named John but with different aliases.

Also this makes users without `name` attribute (service users or guest account) "invisible" for this module. Please, rework this to use `alias` instead as [Zabbix API documentation](https://www.zabbix.com/documentation/4.0/manual/api/reference/user/object) states that it is the required attribute for user object.